src/core/control/tools/EditSelection.cpp

```diff
@@ -80,7 +80,7 @@ EditSelection::EditSelection(UndoRedoHandler* undo, Selection* selection, XojPag
         this->sourceLayer->removeElement(e, false);
     }
 
-    view->rerenderPage();
+    view->rerenderRect(snappedBounds.x, snappedBounds.y, snappedBounds.width, snappedBounds.height);
```

Comment:
That won't quite work: The snappedBounds of Stroke elements correspond to the bounding box of the stroke if it had width == 0 (it is used to snap object to the grid). So in this instance, it'll be too small a rerender.
The simplest thing would possibly be to have `calcSizeFromElements()` return the `Range range` it uses in its computations, and use this returned Range to rerender.